### PR TITLE
fix: use dropped spans outcome in spanmetrics aggregation

### PR DESCRIPTION
**Pull request by kruskall**
## Motivation/summary

Use the correct outcome when creating the aggregation keys: use the outcome from dropped span stats instead of the current event outcome

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## Related issues

Closes https://github.com/elastic/apm-server/issues/10543

### CODE REVIEW

File: changelogs/8.7.asciidoc [hunk 1]
```diff
@@ -86,4 +87,4 @@ https://github.com/elastic/apm-server/compare/v8.7.0\...v8.7.1[View commits]
 - Add overflow metrics collection {pull}10330[10330]
 - Improve APM UI query performance by adding an always enabled service summary metrics aggregator which produces `1m`, `10m` and `60m` aggregation intervals. Store them into data streams `metrics-apm.service_summary.${interval}` and make them hidden for
 non-default intervals {pull}10029[10029] {pull}10060[10060] {pull}10070[10070]
-- Add `service.language.name` {pull}10111[10111]
\ No newline at end of file
+- Add `service.language.name` {pull}10111[10111]
```
Comment (carsonip) [inline changelogs/8.7.asciidoc:90]: What changed on this line? In the diff they look kind of identical

**Review by carsonip — APPROVED** [review]

File: changelogs/8.7.asciidoc (hunk 1 above)
Comment (marclop) [reply changelogs/8.7.asciidoc:90]: double dash

**Review by marclop — APPROVED** [review]

File: changelogs/8.7.asciidoc (hunk 1 above)
Comment (kruskall) [reply changelogs/8.7.asciidoc:90]: this is vim fixing the file by automatically adding an EOL (see https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)

Comment (marclop) [reply changelogs/8.7.asciidoc:90]: oh I confused the `-` diff for a dash 🤦🏼

**Review by carsonip — APPROVED** [review]

**Comment by lahsivjar** [comment]
Tested with `8.7.1`.

<details>
<summary>Input NDJSON</summary>

```
{"metadata": {"service": {"name": "test-service-10592","node": {"configured_name": "chatty-node"},"version": "5.1.3","environment": "staging","language": {"name": "ecmascript","version": "8"},"runtime": {"name": "node","version": "8.0.0"},"framework": {"name": "Express","version": "1.2.3"},"agent": {"name": "elastic-node", "activation_method": "some_activation_method","version": "3.14.0"}},"user": {"id": "123user", "username": "bar", "email": "bar@user.com"}, "labels": {"tag0": null, "tag1": "one", "tag2": 2}, "process": {"pid": 1234,"ppid": 6789,"title": "node","argv": ["node","server.js"]},"system": {"hostname": "prod1.example.com","architecture": "x64","platform": "darwin", "container": {"id": "container-id"}, "kubernetes": {"namespace": "namespace1", "pod": {"uid": "pod-uid", "name": "pod-name"}, "node": {"name": "node-name"}}},"cloud":{"account":{"id":"account_id","name":"account_name"},"availability_zone":"cloud_availability_zone","instance":{"id":"instance_id","name":"instance_name"},"machine":{"type":"machine_type"},"project":{"id":"project_id","name":"project_name"},"provider":"cloud_provider","region":"cloud_region","service":{"name":"lambda"}}}}
{"transaction":{"id":"ddf109a4c4aa5f2b6e984548ca57774c","trace_id":"646df3b8b5279e982cc12a2f1ac004f3","name":"GET /api/types","type":"request","duration":32.592981,"outcome":"success","result":"success","timestamp":1496170407154000,"sampled":true,"span_count":{"started":20,"dropped":3},"context":{"service":{"runtime":{"version":"7.0"}},"page":{"referer":"http://localhost:8000/test/e2e/","url":"http://localhost:8000/test/e2e/general-usecase/"},"request":{"socket":{"remote_address":"12.53.12.1","encrypted":true},"http_version":"1.1","method":"POST","url":{"protocol":"https:","full":"https://www.example.com/p/a/t/h?query=string#hash","hostname":"www.example.com","port":"8080","pathname":"/p/a/t/h","search":"?query=string","hash":"#hash","raw":"/p/a/t/h?query=string#hash"},"headers":{"user-agent":["Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36","Mozilla Chrome Edge"],"content-type":"text/html","cookie":"c1=v1, c2=v2","some-other-header":"foo","array":["foo","bar","baz"]},"cookies":{"c1":"v1","c2":"v2"},"env":{"SERVER_SOFTWARE":"nginx","GATEWAY_INTERFACE":"CGI/1.1"},"body":{"str":"hello world","additional":{"foo":{},"bar":123,"req":"additional information"}}},"response":{"status_code":200,"headers":{"content-type":"application/json"},"headers_sent":true,"finished":true,"transfer_size":258,"encoded_body_size":2690,"decoded_body_size":2990},"user":{"domain":"ldap://abc","id":"99","username":"foo"},"tags":{"organization_uuid":"9f0e9d64-c185-4d21-a6f4-4673ed561ec8","tag2":12,"tag3":12.45,"tag4":false,"tag5":null},"custom":{"my_key":1,"some_other_value":"foo bar","and_objects":{"foo":["bar","baz"]},"(":"not a valid regex and that is fine"}},"dropped_spans_stats":[{"type":"request","subtype":"http","destination_service_resource":"example.com:443","outcome":"failure","duration":{"count":2,"sum":{"us":123456}}},{"type":"db","subtype":"mysql","destination_service_resource":"mysql","outcome":"success","duration":{"count":1,"sum":{"us":9876543}}}]}}
```
</details>

<details>
<summary>Output Assertion</summary>

Screenshot 2023-04-27 at 4 31 41 PM

</details>

**Pull request merged**